### PR TITLE
Submit as JCL: check for profile before presenting quick pick

### PR DESCRIPTION
**Pull request by likhithanimma1**
## Proposed changes
Fixes # #2628
Since we store temp. files based on their profile, we could determine the profile without spawning a quick pick for user input.

## Release Notes

Milestone:

Changelog:

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Review by JillieBeanSim — COMMENTED** [review]
This looks good for this workflow, thanks @likhithanimma1 for the update.  Would you mind updating the CHANGELOG please?

**Review by JillieBeanSim — DISMISSED** [review]

File: packages/zowe-explorer/src/dataset/actions.ts [hunk 1]
```diff
@@ -931,20 +931,28 @@ export async function submitJcl(datasetProvider: api.IZoweTree<api.IZoweDatasetT
...
+                    return;
+                }
+            } else if (profileNamesList.length > 0) {
+                sessProfileName = profileNamesList[0];
+            } else {
+                api.Gui.showMessage(localize("submitJcl.noProfile", "No profiles available"));
             }
         } else {
-            api.Gui.showMessage(localize("submitJcl.noProfile", "No profiles available"));
+            // eslint-disable-next-line no-useless-escape
```
Comment (traeok) [inline packages/zowe-explorer/src/dataset/actions.ts:953]: This linter exception can be removed since the RegEx expression was replaced 👍

**Review by traeok — COMMENTED** [review]
Tested and LGTM - just noticed an outdated ESLint comment during review, will approve once removed 🙂 
Thanks @likhithanimma1!

File: packages/zowe-explorer/src/dataset/actions.ts (hunk 1 above)
Comment (likhithanimma1) [reply packages/zowe-explorer/src/dataset/actions.ts:953]: Hey thanks Trae for noticing that

**Review by traeok — APPROVED** [review]

File: packages/zowe-explorer/src/dataset/actions.ts [hunk 2]
```diff
@@ -931,20 +931,27 @@ export async function submitJcl(datasetProvider: api.IZoweTree<api.IZoweDatasetT
...
+                    ignoreFocusOut: true,
+                    canPickMany: false,
+                };
+                sessProfileName = await api.Gui.showQuickPick(profileNamesList, quickPickOptions);
+                if (!sessProfileName) {
+                    api.Gui.infoMessage(localizedStrings.opCancelled);
+                    return;
+                }
+            } else if (profileNamesList.length > 0) {
+                sessProfileName = profileNamesList[0];
```
Comment (zFernand0) [inline packages/zowe-explorer/src/dataset/actions.ts:948]: This is really cool! 😋 
Should we do the same for other types of profiles across Zowe Explorer?

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋

**Pull request merged**